homeassistant/components/withings/common.py

```diff
@@ -119,357 +140,406 @@ class StateData:
-        GetSleepSummaryField.SNORING,
-        "Snoring",
-        "",
-        None,
-        SENSOR_DOMAIN,
-        False,
-        UpdateType.POLL,
+    WithingsSensorEntityDescription(
+        measurement=Measurement.SLEEP_SNORING,
+        measure_type=GetSleepSummaryField.SNORING,
+        name="Snoring",
+        platform=SENSOR_DOMAIN,
+        update_type=UpdateType.POLL,
+        key="Snoring",
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.MEASUREMENT,
     ),
-    WithingsAttribute(
-        Measurement.SLEEP_SNORING_EPISODE_COUNT,
-        GetSleepSummaryField.SNORING_EPISODE_COUNT,
-        "Snoring episode count",
-        "",
-        None,
-        SENSOR_DOMAIN,
-        False,
-        UpdateType.POLL,
+    WithingsSensorEntityDescription(
+        measurement=Measurement.SLEEP_SNORING_EPISODE_COUNT,
+        measure_type=GetSleepSummaryField.SNORING_EPISODE_COUNT,
+        name="Snoring episode count",
+        platform=SENSOR_DOMAIN,
+        update_type=UpdateType.POLL,
+        key="Snoring episode count",
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.MEASUREMENT,
     ),
-    WithingsAttribute(
-        Measurement.SLEEP_WAKEUP_COUNT,
-        GetSleepSummaryField.WAKEUP_COUNT,
-        "Wakeup count",
-        const.UOM_FREQUENCY,
-        "mdi:sleep-off",
-        SENSOR_DOMAIN,
-        False,
-        UpdateType.POLL,
+    WithingsSensorEntityDescription(
+        measurement=Measurement.SLEEP_WAKEUP_COUNT,
+        measure_type=GetSleepSummaryField.WAKEUP_COUNT,
+        name="Wakeup count",
+        platform=SENSOR_DOMAIN,
+        update_type=UpdateType.POLL,
+        key="Wakeup count",
+        native_unit_of_measurement=const.UOM_FREQUENCY,
+        icon="mdi:sleep-off",
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.MEASUREMENT,
     ),
-    WithingsAttribute(
-        Measurement.SLEEP_WAKEUP_DURATION_SECONDS,
-        GetSleepSummaryField.WAKEUP_DURATION,
-        "Wakeup time",
-        TIME_SECONDS,
-        "mdi:sleep-off",
-        SENSOR_DOMAIN,
-        False,
-        UpdateType.POLL,
+    WithingsSensorEntityDescription(
+        measurement=Measurement.SLEEP_WAKEUP_DURATION_SECONDS,
+        measure_type=GetSleepSummaryField.WAKEUP_DURATION,
+        name="Wakeup time",
+        platform=SENSOR_DOMAIN,
+        update_type=UpdateType.POLL,
+        key="Wakeup time",
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        icon="mdi:sleep-off",
+        device_class=SensorDeviceClass.DURATION,
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.MEASUREMENT,
     ),
     # Webhook measurements.
-    WithingsAttribute(
-        Measurement.IN_BED,
-        NotifyAppli.BED_IN,
-        "In bed",
-        "",
-        "mdi:bed",
-        BINARY_SENSOR_DOMAIN,
-        True,
-        UpdateType.WEBHOOK,
+    WithingsBinarySensorEntityDescription(
```

Comment:
Entity descriptions from different platforms don't belong in the same container. They are platform details and should really just be defined in each platform.

Reply:
Aaargh.
This integration will need to be unpickled!
Step 1 => #84089